projects/core/services/position.service.ts

```diff
@@ -7,20 +7,17 @@ import {finalize, map, Observable} from 'rxjs';
 
 @Injectable()
 export class TuiPositionService extends Observable<TuiPoint> {
-    constructor(
-        // Destructuring here causes memory leak
-        @Inject(ElementRef) el: ElementRef<HTMLElement>,
-        @Inject(ANIMATION_FRAME) animationFrame: Observable<unknown>,
-        @Inject(NgZone) zone: NgZone,
-        @Inject(TuiPositionAccessor) accessor: TuiPositionAccessor,
-    ) {
+    private readonly el: HTMLElement = inject(ElementRef).nativeElement;
+    private readonly accessor = inject(TuiPositionAccessor);
+
+    constructor() {
         super(subscriber =>
-            animationFrame
+            inject(ANIMATION_FRAME)
```

Comment:
This wouldn't work, this function will not be called in constructor context. Let's make a `private readonly stream$` that we will use here, like we do in other similar places.

Reply:
fixed